src/clj/rems/application/cleaner.clj

```diff
@@ -0,0 +1,57 @@
+(ns rems.application.cleaner
+  (:require [mount.core :as mount]
+            [clj-time.core :as time]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [rems.scheduler :as scheduler]
+            [rems.db.applications :as applications]
+            [rems.config :refer [env]]))
+
+(defn- is-expired-draft?
+  [application]
+  (let [expiration-threshold (:draft-application-expiration-threshold env)
+        last-activity (:application/last-activity application)
+        state (:application/state application)]
+    (and (= :application.state/draft state)
+         (time/before? last-activity (time/minus (time/now) (time/days expiration-threshold))))))
+
+(deftest test-is-expired-draft
+  (with-redefs [env {:draft-application-expiration-threshold 90}]
+    (testing "should identify expired draft application"
+      (are [expected input] (= expected (is-expired-draft? input))
+        true {:application/state :application.state/draft
+              :application/last-activity (time/minus (time/now) (time/days 90) (time/seconds 1))}
+        false {:application/state :application.state/draft
+               :application/last-activity (time/now)}
+        false {:application/state :application.state/submitted
+               :application/last-activity (time/minus (time/now) (time/days 90) (time/seconds 1))}))))
+
+(defn process-expired-draft-applications! []
+  (let [removed-ids (atom [])
+        failed-ids (atom [])]
+    (doseq [app-id (->> (applications/get-all-unrestricted-applications)
+                        (filter is-expired-draft?)
+                        (map :application/id))]
+      (try
+        (applications/remove-application-data! app-id)
+        (swap! removed-ids conj app-id)
+        (catch Throwable t
+          (log/error (str "Removing application with id " app-id " failed: ") t)
+          (swap! failed-ids conj app-id))))
+    (when (seq @failed-ids)
+      (log/info "process-expired-draft-applications! failed to remove applications" @failed-ids))
+    (when (seq @removed-ids)
+      (log/info "process-expired-draft-applications! succesfully removed applications" @removed-ids))
```

Comment:
Maybe we don't need this "complex logic"? I mean there is no harm in logging once for each application id. This would eliminate the need for the atoms. Also how about making it `log/warn` or `log/error` here because it is a problem somewhere that needs fixing?